src/190B20.c

```diff
@@ -1640,7 +1705,14 @@ INCLUDE_ASM(s32, "190B20", btl_appendGfx_prim_quad);
 //     gDPSetCombineMode(gMasterGfxPos++, G_CC_DECALRGBA, G_CC_DECALRGBA);
 // }
 
-INCLUDE_ASM(s32, "190B20", btl_draw_prim_quad);
+void btl_draw_prim_quad(s32 arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s32 arg6, s32 arg7) {
+    u16 new_var;
+    u16 new_var2;
+
+    new_var = (arg4 + arg6) & 0xFFFF;
+    new_var2 = (arg5 + arg7) & 0xFFFF;
+    btl_appendGfx_prim_quad(arg0 & 0xFF, arg1 & 0xFF, arg2 & 0xFF, arg3 & 0xFF, arg4 & 0xFFFF, arg5 & 0xFFFF, new_var, new_var2);
```

Comment:
These 0xFFs are most likely u8 casts, and the 0xFFFFs u16 casts, so the ideal thing to do here is to find btl_appendGfx_prim_quad, and modify its signature accordingly to be u8, u8, u8, u8, u16, u16, u16, u16 and then add a declaration for it in functions.h if that function doesn't exist in this file. If the function isn't decomped, you'll need to modify the INCLUDE_ASM line for it, and that's kind of weird since it's a macro. let us know if you'd like help in discord

after this, you should be able to remove al the & 0xFF(FF)s